### ISSUE TITLE
Remove unnecessary prefix / suffix in store keys

### ISSUE THREAD

**Issue by robert-zaremba** [umee-network/umee#1581]
In https://github.com/umee-network/umee/pull/1578 we cleaned up store key constructors.
Various key have unnecessary length prefix or null terminator. Both are not needed if it is the last and ultimate part of the key.
Example: https://github.com/umee-network/umee/pull/1578/files#diff-b3ff0f6b42fcc3132cbb1a0210e624410a4e1ca4c7027c70928edad4ca1880baR38

It's nice to have, and we need to guarantee that the keys won't be extended.

**Comment by robert-zaremba** [comment]
Let's close this. It will be solved by #1668 and https://github.com/umee-network/umee/pull/1656

**Comment by robert-zaremba** [comment]
up

**Comment by robert-zaremba** [comment]
@zarazan or @rbajollari do you think you can handle this?

**Comment by toteki** [comment]
If the keys are currently working and consistent with each other, we should avoid unnecessary changes due to
- requiring store migrations (delete old, add new, for all stored values)
- unintentionally introducing bugs (removing a null terminator from a `prefix | denom | 0x00` could cause hypothetical denom `uumee2` to be caught up in iterators over `uumee`)

TLDR avoid unnecessary changes
